library/src/main/java/com/datatorrent/lib/io/fs/HDFSOutputOperator.java

```diff
@@ -0,0 +1,393 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.io.fs;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.StreamCodec;
+import com.datatorrent.netlet.util.DTThrowable;
+
+/**
+ * This class is responsible for writing tuples to HDFS. All tuples are written
+ * to the same file. Rolling file based on size, no. of tuples, idle windows,
+ * elapsed windows is supported.
+ * 
+ * @param <T>
+ */
+
+class HDFSOutputOperator extends AbstractFileOutputOperator<byte[]>
+{
+
+  /**
+   * Name of the file to write the output. Directory for the output file should
+   * be specified in the filePath
+   */
+  @NotNull
+  private String fileName;
+
+  /**
+   * currentPartName is of the form fileName_physicalPartionId.partNumber
```

Comment:
I guess it can be of form "fileName_physicalPartionId.partNumber" (or it's a default), user can give any other format right?